### PR TITLE
WP Stories integration - Media Picker part 2: using EditorMedia to source from the picker

### PR DESCRIPTION
**Pull request by mzorz**
This PR builds on top of #11939

This PR implements the changes needed for StoryComposerActivity to bridge its interactions with the MediaPicker through EditorMedia, in the same way the EditPostActivity works.

This PR introduces a change in EditorMedia to be able to tell its use cases whether upload is optional or necessary. Up til now, we were enqueueing images added to a Post immediately for upload, but with Stories it's a little bit different, as we only want to upload the composed images (images plus text and emoji) in the Post. The EditorMedia encapsulates the different problems from taking media from different sources and making them available to an Editor very well so, it makes sense to follow the same paths and only diverge in the upload part.

We introduced then, a flag called
```
enum class EditorType {
    POST_EDITOR,
    STORY_EDITOR
}
```
- If EditorMedia is being used in the context of an editor of type `POST_EDITOR`, then its flow will include uploading, as normally.
- If EditorMedia is being used in the context of an editor of type `STORY_EDITOR`, then upload is not made part of the flow, as simple as that.

We will split the upload part later, once the Story frames have been locally saved and are available to be actually uploaded to the backend.

Note for now this only has the media from the device path covered (to keep PRs simple). We'll work on WPMedia and Free Photo source paths on subsequent PRs.

To test:
1. run the app
2. select a site
3. tap the FAB button on the "MySite" screen
4. select Story
5. observe the Story creation experience appears (camera preview)
6. tap on the upload icon to the left of the camera shutter
7. observe the WPAndroid media picker gets launched
8. select some media items and tap on the checkmark on the top toolbar to indicate you're done selecting
9. observe these media items are added as background for as many Story pages as media items were selected.

Also test:

0. fresh install
1. repeat steps 1-8 above
2. the image optimisation dialog should appear. Verify it appears correctly and the app doesn't crash (this was saved by applying the correct style in 12ca7d3)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -220,7 +220,7 @@
             android:name=".ui.stories.StoryComposerActivity"
             android:label="@string/app_name"
             android:screenOrientation="portrait"
-            android:theme="@style/AppTheme.Immersive">
+            android:theme="@style/WordPress.Stories.Immersive">
```
Comment (mzorz) [inline WordPress/src/main/AndroidManifest.xml:223]: this is a seemingly unrelated change - the action bar looked weird when triggering the media picker from the story composer, that's why it's here (we have had this theme already applied in the demo app correspondingly, although differently named)

File: WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt [hunk 2]
```diff
@@ -49,8 +129,96 @@ class StoryComposerActivity : ComposeLoopFrameActivity(), SnackbarProvider, Medi
...
+        editorMedia.toastMessage.observe(this,
+                Observer<Event<ToastMessageHolder>> { event: Event<ToastMessageHolder?> ->
+                    val contentIfNotHandled = event.getContentIfNotHandled()
+                    contentIfNotHandled?.show(this)
+                }
+        )
+    }
+
+    // EditorMediaListener
+    override fun appendMediaFiles(mediaFiles: Map<String, MediaFile>) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt:192]: As mentioned above, we might want to consider moving all this logic into a VM. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt [hunk 3]
```diff
@@ -1,21 +1,74 @@
...
+class StoryComposerActivity : ComposeLoopFrameActivity(), SnackbarProvider, MediaPickerProvider, EditorMediaListener {
     private var site: SiteModel? = null
 
+    @Inject lateinit var editorMedia: EditorMedia
+    @Inject lateinit var progressDialogHelper: ProgressDialogHelper
+    @Inject lateinit var uiHelpers: UiHelpers
+    @Inject lateinit var postStore: PostStore
+    private var addingMediaToEditorProgressDialog: ProgressDialog? = null
+
+    companion object {
```
[the remark this answers is not in the document]
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt:53]: totally test code so we could have a PostModel of the sorts ready and easily accessible :D - all of this is changed later in this PR #12031  - thanks for pointing that out!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt [hunk 4]
```diff
@@ -126,7 +134,8 @@ class EditorMedia @Inject constructor(
                     uriList,
                     site,
                     freshlyTaken,
-                    editorMediaListener
+                    editorMediaListener,
+                    (editorType == POST_EDITOR) // also start upload if this is a Posts editor
```
[the remark this answers is not in the document]
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt:138]: Yes :) it was added in this PR https://github.com/wordpress-mobile/WordPress-Android/pull/11972, merged into this one, can be seen here https://github.com/wordpress-mobile/WordPress-Android/pull/11953/files#diff-309cf9193e3dc7820946c72162bc9ad8R39-R90

**Comment by mzorz** [comment]
> I've reviewed the code and left some comments. Most of them are related to moving some business logic away from the view layer. It'd be also great if the logic was tested. I haven't run the app yet as my comments my result in some changes. Let me know what you think and I'll test it after ;). Thanks

Thank you @malinajirka ! I've been considering this and I think, given this already introduces a new path in the app that tries to mimick EditPostActivity as explained in the PR description - I'd prefer to keep this one as is (and be easily comparable to EditPostActivity as it currently is), and then make the suggested changes on a different PR, ideally after the rest of the functionality gets merged for our first alpha (it'd be an internal test build).

I added an item to my ToDo list for our second alpha to include this refactor 👍 - let me know if you think that's ok? 🙏 thanks

**Comment by mzorz** [comment]
>  If we make sure that we refactor the code as part of this initiative, I'm completely fine merging it. I'd just like to avoid adding tech debt into our code base just to keep it consistent with EditPostActivity (which we all known isn't in an ideal state).

Thanks a lot! Yes totally agreed on this, as I said we added it to our ToDo list for the second alpha, for which we already had planned other refactors to introduce more testability and such, so it fits that phase's focus perfectly 👍 .

**Comment by mzorz** [comment]
Comments have been tracked elsewhere to be tackled on second alpha, PRs are approved,  merging this one
